R/collection_extract.R

```diff
@@ -0,0 +1,138 @@
+#' Given a (multi)geometry, return a (multi)geometry consisting only of elements of the specified type.
+#'
+#' Similar to st_collection_extract in postgis. If there are no sub-geometries of the specified type, an empty geometry is returned.
+#'
+#' @param x an \code{sf(c/g)} object that has mixed geometry
```

Comment:
an object of class \code{sf}, \code{sfc} or \code{sfg} that has...

With mixed geometry, do you also mean MULTIPOLYGON or only GEOMETRYCOLLECTION?

Reply:
As it's set up it works on only `GEOMETRYCOLLECTION` for `sfg` and `GEOMETRY` or `GEOMETRYCOLLECTION` for `sfc` and `sf`. It doesn't work to extract `POLYGON` from `MULTIPOLYGON` etc. I *think* this is consistent with `ST_CollectionExtract`. I'll try to make that clearer in the documentation.